src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -439,16 +444,25 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         typeSpec: TypeSpec = anonymousClassBuilder("").build()) = apply {
       check(kind == Kind.ENUM) { "${this.name} is not enum" }
       require(typeSpec.anonymousTypeArguments != null) {
-          "enum constants must have anonymous type arguments" }
+          "enum constants must have anonymous type arguments"
+      }
       require(name.isName) { "not a valid enum constant: $name" }
       enumConstants.put(name, typeSpec)
     }
 
     fun addProperties(propertySpecs: Iterable<PropertySpec>) = apply {
-      this.propertySpecs += propertySpecs
+      propertySpecs.map(this::addProperty)
     }
 
     fun addProperty(propertySpec: PropertySpec) = apply {
+      if (kind == Kind.HEADER_CLASS) {
+        require(propertySpec.initializer.isNullOrEmpty()) {
```

Comment:
Is a null check sufficient? Is it ever empty? (Or vice-versa!)

Reply:
Currently `initializer` can be both, since it's optional and we don't prohibit passing in an empty `CodeBlock`. I'll see if there are valid use-cases for the latter, and if not - I'd go with disallowing an empty `CodeBlock` as `initializer` in `PropertySpec`.

Reply:
Actually, we don't want any initializers, so I guess a null-check is sufficient. Pushed cb3c5e4